### ISSUE TITLE
Update trace-viewer’s importing code to follow the tracing clock sync architecture

### ISSUE THREAD

**Issue by wangzhen127** [catapult-project/catapult#1606]
We need to update the importer code to follow the tracing clock sync architecture.

The master bug: https://crbug.com/534578
This is blocked on https://crbug.com/542390

@zeptonaut @wangzhen127

**Comment by wangzhen127** [comment]
@zeptonaut , I assigned the bug to you. But I would also love to help. :)

**Comment by wangzhen127** [comment]
@nedn

**Comment by nedn** [comment]
Design doc: https://docs.google.com/document/d/11qp-38vI-aFU03i8YpJlvJwKgectugVjPysf27Ea8js/edit#

**Comment by zeptonaut** [comment]
It looks like ETW also has some clock sync functionality: [link](https://github.com/catapult-project/catapult/blob/4884564bbfac31b82d82bbe68217d59e2fcad8f7/tracing/tracing/extras/importer/etw/etw_importer.html#L353)

**Comment by wangzhen127** [comment]
Does that indicate there is some implicit clock sync for ETW going on?

**Comment by zeptonaut** [comment]
It looks like it's explicit. Some ETW event looks like:

```javascript
{
  guid: ...,
  walltime: 1234,    // The time of the event according to the ETW clock
  ticks: 5678        // The time of the event according to the QueryPerformanceCounter
}
```

The ETW importer then shifts all other events according to the difference between the two.

**Comment by nedn** [comment]
So those event aren't used for syncing ETW's clock against the controller's clock but for syncing its clock against QueryPerformanceCounter clock?

**Comment by zeptonaut** [comment]
QueryPerformanceCounter is the high-res monotonic clock that Chrome uses on Windows, so I think that syncing against QPC and syncing against the controller's clock are the same in this instance.

**Comment by nedn** [comment]
Correct. But I think that code also need to be updated, for cases like telemetry/systrace being the controller and not Chrome.

**Comment by zeptonaut** [comment]
Agreed. I have it on my list of TODOs (after the ftrace importer and the BattOr importer).

**Comment by zeptonaut** [comment]
Actually, leaving this issue open: I still need to move ETW to use the new clock sync marker format.
